cmd/nexusUpload.go

```diff
@@ -126,19 +133,50 @@ func nexusUpload(options nexusUploadOptions, _ *telemetry.CustomData) {
 }
 
 func runNexusUpload(utils nexusUploadUtils, uploader nexus.Uploader, options *nexusUploadOptions) error {
-	err := uploader.SetRepoURL(options.Url, options.Version, options.Repository)
+	performMavenUpload := len(options.MavenRepository) > 0
+	performNpmUpload := len(options.NpmRepository) > 0
+	err := uploader.SetRepoURL(options.Url, options.Version, options.MavenRepository, options.NpmRepository)
+	if err != nil {
+		return err
+	}
+
+	if utils.usesNpm() && performNpmUpload {
+		log.Entry().Info("NPM project structure detected")
+		err = uploadNpmArtifacts(utils, uploader, options)
+	} else {
+		log.Entry().Info("Skipping npm upload because either no package json was found or NpmRepository option is not provided.")
+	}
 	if err != nil {
 		return err
 	}
-	if utils.usesMta() {
-		log.Entry().Info("MTA project structure detected")
-		return uploadMTA(utils, uploader, options)
-	} else if utils.usesMaven() {
-		log.Entry().Info("Maven project structure detected")
-		return uploadMaven(utils, uploader, options)
+
+	if performMavenUpload {
```

Comment:
Why here just performMavenUpload  and not something like performMavenUpload  && usesMaven() ?

Reply:
sorry. seen below :-)